truss-chains/truss_chains/remote.py

```diff
@@ -437,7 +359,64 @@ def push(
     non_entrypoint_root_dir: Optional[str] = None,
     gen_root: pathlib.Path = pathlib.Path(tempfile.gettempdir()),
 ) -> Optional[ChainService]:
-    return _Pusher(options, gen_root).push(entrypoint, non_entrypoint_root_dir)
+    chainlet_artifacts = _Generator(options, gen_root).generate_chainlet_artifacts(
+        entrypoint,
+        non_entrypoint_root_dir,
+    )
+
+    if options.only_generate_trusses:
+        return None
+
+    if isinstance(options, definitions.PushOptionsBaseten):
+        _create_chains_secret_if_missing(options.remote_provider)
+
+        return _create_baseten_chain(options, chainlet_artifacts)
+    elif isinstance(options, definitions.PushOptionsLocalDocker):
+        entrypoint_service: Optional[DockerTrussService] = None
+        chainlet_to_predict_url: Dict[str, Dict[str, str]] = {}
+        chainlet_to_port: Dict[str, int] = {}
+
+        for chainlet_artifact in chainlet_artifacts:
+            port = utils.get_free_port()
+            chainlet_to_port[chainlet_artifact.name] = port
+
+            # http://localhost:{port} seems to only work *sometimes* with docker.
+            predict_url = f"http://host.docker.internal:{port}"
+
+            chainlet_to_predict_url[chainlet_artifact.name] = {
+                "predict_url": predict_url,
+            }
+
+            if chainlet_artifact.is_entrypoint:
+                assert entrypoint_service is None
+
+                entrypoint_service = DockerTrussService(predict_url, is_draft=True)
+
+        assert entrypoint_service is not None
+
+        local_config_handler.LocalConfigHandler.set_dynamic_config(
+            definitions.DYNAMIC_CHAINLET_CONFIG_KEY,
+            json.dumps(chainlet_to_predict_url),
+        )
```

Comment:
Interesting find, I dind't know about this before! Does this get implicitly picked up by `truss_handle.docker_run` later?

Reply:
I believe so: https://github.com/basetenlabs/truss/blob/3bf38ef47d85cea51d5dcb5c4e15aaa4c8d21572/truss/truss_handle/truss_handle.py#L310-L315

Reply:
Verified this works via integration testing.